src/Filesystem/Provider/FileAccess.php

```diff
@@ -0,0 +1,257 @@
+	 * @version 1.0
+	 */
+	public function delete($path);
+
+
+	/**
+	 * Reads the entire file content into a string and removes the file afterwards.
+	 *
+	 * @param string $path  The file which should be red and removed.
+	 *
+	 * @return string       The entire file content.
+	 *
+	 * @throws FileNotFoundException    If the file was not found.
+	 * @throws IOException              If the file could not red or deleted.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function readAndDelete($path);
+
+
+	/**
+	 * Moves a file from the source to the destination.
+	 *
+	 * @param string $path      The current path of the file which should be moved.
+	 * @param string $newPath   The new path of the file.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException        If the source file is not found.
+	 * @throws FileAlreadyExistsException   If the destination file is already existing.
+	 * @throws IOException                  If the file could not be moved.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function rename($path, $newPath);
+
+
+	/**
+	 * Copy the source file to a destination.
+	 *
+	 * @param string $path      The source path to the file which should be copied.
+	 * @param string $copyPath  The destination path of the file copy.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException        If the source file does not exist.
+	 * @throws FileAlreadyExistsException   If the destination file already exists.
+	 * @throws IOException                  If the file could not be copied to the destination.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function copy($path, $copyPath);
+
+
+	/**
+	 * Get a files mime-type.
+	 *
+	 * @param string $path  The file which should be used to get the mime-type.
+	 *
+	 * @return string   The mime-type of the file.
+	 *
+	 * @throws FileNotFoundException    If the file is not found.
+	 * @throws IOException              If the mime-type could not be determined.
+	 */
+	public function getMimeType($path);
+
+
+	/**
+	 * Get the timestamp of the file.
+	 *
+	 * @param string $path  The path to the file.
+	 *
+	 * @return \DateTime  The timestamp of the file.
+	 *
+	 * @throws FileNotFoundException    If the file is not found.
+	 * @throws IOException              If the file can not be red.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function getTimestamp($path);
+
+
+	/**
+	 * Get the size of a file.
+	 *
+	 * The file size units are provided by the FileSize class.
+	 *
+	 * @param string $path         The path to the file.
+	 * @param int    $fileSizeUnit The unit of the file size, which are defined in the FileSize class.
+	 *
+	 * @return FileSize
+	 * @since   5.3
+	 * @version 1.0
+	 *
+	 * @see FileSize
+	 */
+	public function getSize($path, $fileSizeUnit);
+
+	/**
+	 * Sets the visibility for a file.
+	 * Please note that the $visibility must 'public' or 'private'.
+	 *
+	 * The Visibility interface provides two constants PUBLIC_ACCESS and PRIVATE_ACCESS.
```

Comment:
I'd like to make this a MUST:

`Consumers of the API MUST use PUBLIC_ACCESS or PRIVATE_ACCESS`